jsonrpc/jsonrpc.go

```diff
@@ -220,7 +225,14 @@ func (j *JSONRPC) handle(w http.ResponseWriter, req *http.Request) {
 
 	resp, err := j.dispatcher.Handle(data)
 	if err != nil {
-		handleErr(err)
+		id := err.(*ErrorObject).Data
+		err.(*ErrorObject).Data = nil
+		errorResponse := ErrorResponse{
+			JSONRPC: "2.0",
+			ID:      id,
+			Error:   err.(*ErrorObject),
+		}
+		handleErr(errorResponse)
```

Comment:
Why is the actual error being unwrapped here, modified, then being put back? What is the purpose of setting the Data field to nil in the Error field of ErrorResponse?

Reply:
Fixed this